addons/website_sale_wishlist/migrations/12.0.1.0/noupdate_changes.xml

```diff
@@ -0,0 +1,7 @@
+<?xml version='1.0' encoding='utf-8'?>
+<odoo>
+  <record id="product_wishlist_rule" model="ir.rule">
+    <field name="domain_force">["|", ('partner_id','=', user.partner_id.id), "&amp;", ("partner_id", "=", False), ("session", "=", user.current_session)]</field>
+    <field name="groups" eval="[(4, ref('base.group_portal')),(4, ref('base.group_public')), (4, ref('base.group_user'))]"/>
```

Comment:
If you apply the script as is (with the paths correctly sorted), you will get:
```xml
   <field name="groups" eval="[(4, ref('base.group_portal')), (4, ref('base.group_user'))]"/>
```
but this won't make the desire effect, as we want to remove the public group from this rule, and `(4, x)` operations only add without replacing, and thus you need to change the expression to:
```xml
   <field name="groups" eval="[(3, ref('base.group_public')), ]"/>
```

Reply:
an alternative is:
```xml
<field name="groups" eval="[(5,), (4, ref('base.group_portal')), (4, ref('base.group_user'))]"/>
```